frontend/src/components/common/CodeEditor.jsx

```
@@ -0,0 +1,199 @@
+  useEffect(() => {
+    if (monacoRef.current) {
+      setEditorTheme(monacoRef.current, isDarkMode);
+    }
+  }, [isDarkMode]);
+  
+  // Update error markers when they change
+  useEffect(() => {
+    if (editorRef.current && monacoRef.current) {
+      updateErrorDecorations(editorRef.current, monacoRef.current, markers);
+    }
+  }, [markers, updateErrorDecorations]);
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)